mmtrack/apis/test.py

```diff
@@ -7,45 +8,98 @@
 from mmcv.runner import get_dist_info
 
 
 def single_gpu_test(model,
                     data_loader,
                     show=False,
                     out_dir=None,
+                    fps=3,
                     show_score_thr=0.3):
     """Test model with single gpu.
 
     Args:
         model (nn.Module): Model to be tested.
         data_loader (nn.Dataloader): Pytorch data loader.
-        show (bool): If True, visualize the prediction results (Not supported
-            for now).  Defaults to False.
-        out_dir (str): Path of directory to save the visualization results (Not
-            supported for now). Defaults to None.
-        show_score_thr (float): The score threthold of visualization (Not
-            supported for now). Defaults to 0.3.
+        show (bool, optional): If True, visualize the prediction results.
+            Defaults to False.
+        out_dir (str, optional): Path of directory to save the
+            visualization results. Defaults to None.
+        fps (int, optional): FPS of the output video.
+            Defaults to 3.
+        show_score_thr (float, optional): The score threshold of visualization.
+            Defaults to 0.3.
 
     Returns:
         dict[str, list]: The prediction results.
     """
     model.eval()
     results = defaultdict(list)
     dataset = data_loader.dataset
+    last_img_meta = None
     prog_bar = mmcv.ProgressBar(len(dataset))
     for i, data in enumerate(data_loader):
         with torch.no_grad():
             result = model(return_loss=False, rescale=True, **data)
         for k, v in result.items():
             results[k].append(v)
 
+        batch_size = data['img'][0].size(0)
         if show or out_dir:
-            pass  # TODO
+            assert batch_size == 1, 'Only support batch_size=1 when testing.'
+            img_tensor = data['img'][0]
+            img_meta = data['img_metas'][0].data[0][0]
+            img = tensor2imgs(img_tensor, **img_meta['img_norm_cfg'])[0]
+
+            h, w, _ = img_meta['img_shape']
+            img_show = img[:h, :w, :]
+
+            ori_h, ori_w = img_meta['ori_shape'][:-1]
+            img_show = mmcv.imresize(img_show, (ori_w, ori_h))
+
+            if out_dir:
+                out_file = osp.join(out_dir, img_meta['ori_filename'])
+            else:
+                out_file = None
+
+            model.module.show_result(
+                img_show,
+                result,
+                show=show,
+                out_file=out_file,
+                score_thr=show_score_thr)
+
+            # generate the video from images
+            # case1: the current frame is the first frame of the new video,
+            # except the first frame of the entire dataset.
+            # case2: the current frame is the last frame of the entire dataset.
+            if i != 0 and img_meta['frame_id'] == 0 or i == len(dataset):
+                img_prefix, img_name = last_img_meta['ori_filename'].rsplit(
```

Comment:
add `prev_` to all variables as perfix under this `if`.
for example, `prev_img_name`